### PR TITLE
Colors and Typography improvements

### PR DESCRIPTION
**Pull request by vmilan**
# Description

Shortcut: [#265838](https://app.shortcut.com/cartoteam/story/265838/review-foundations-colors-and-typography)

Changes are happening as a result of several meetings that we had with the design team to build solid foundations.

## Type of change

- Feature

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -20,11 +20,9 @@ Removed unused custom `createTheme` function in `carto-theme.js`.
 
 Added several custom variants to the typography set. Note that MUI v5 is needed to register them properly.
 
-## Colors
-
-Keys renamed:
+`Open Sans` and `Montserrat` families are been replaced by `Inter` and `Overpass Mono`, you have an exaple of this in the [`preview-head.html`](https://github.com/CartoDB/carto-react/blob/master/packages/react-ui/storybook/.storybook/preview-head.html) file.
```
Comment (VictorVelarde) [inline UPGRADE.md:23]: Little typo
```suggestion
`Open Sans` and `Montserrat` families have been replaced by `Inter` and `Overpass Mono`, you have an example of this in the [`preview-head.html`](https://github.com/CartoDB/carto-react/blob/master/packages/react-ui/storybook/.storybook/preview-head.html) file.
```

File: packages/react-ui/src/theme/carto-theme.js [hunk 2]
```diff
@@ -58,7 +58,7 @@ export const cartoThemeOptions = {
     /* Custom Colors palette */
     qualitative: { ...commonPalette.qualitative },
     default: { ...commonPalette.default },
-    charts: { ...componentsPalette.charts },
+    cartoBrand: { ...commonPalette.cartoBrand },
```
Comment (VictorVelarde) [inline packages/react-ui/src/theme/carto-theme.js:61]: The name looks a bit strange to me... saying 'cartoBrand' inside a carto theme seems a bit redundant, how about just brand?

File: packages/react-ui/src/theme/sections/palette.js [hunk 3]
```diff
@@ -171,29 +171,28 @@ export const commonPalette = {
...
     dark: baseColors.neutral[50],
-    // are we missing here relatedLight ??
+    light: baseColors.neutral[100],
     contrastText: baseColors.common.black,
-    background: baseColors.shades.dark[8]
+    outlinerBorder: baseColors.shades.dark[25],
+    filledBackground: baseColors.shades.dark[4]
+  },
+  cartoBrand: {
+    naviBlue: '#162945',
```
Comment (VictorVelarde) [inline packages/react-ui/src/theme/sections/palette.js:183]: typo?
```suggestion
    navyBlue: '#162945',
```

Comment (VictorVelarde) [reply packages/react-ui/src/theme/sections/palette.js:183]: or is it for navigation?

File: packages/react-ui/storybook/stories/common/Palette.stories.js [hunk 4]
```diff
@@ -142,25 +148,15 @@ const ActionTemplate = () => {
...
-      <ColorBox colorVariant={'grey'} colorName={'400'} />
-      <ColorBox colorVariant={'grey'} colorName={'300'} />
-      <ColorBox colorVariant={'grey'} colorName={'200'} />
-      <ColorBox colorVariant={'grey'} colorName={'100'} />
-      <ColorBox colorVariant={'grey'} colorName={'50'} />
-      <ColorBox colorVariant={'grey'} colorName={'A100'} />
-      <ColorBox colorVariant={'grey'} colorName={'A200'} />
-      <ColorBox colorVariant={'grey'} colorName={'A400'} />
-      <ColorBox colorVariant={'grey'} colorName={'A700'} />
+      <ColorBox colorVariant={'cartoBrand'} colorName={'naviBlue'} />
```
Comment (VictorVelarde) [inline packages/react-ui/storybook/stories/common/Palette.stories.js:156]: If you're ok with naming, I'm ok, it's just that 'cartoBrand' vs cartoTheme feel a bit like strange in the naming

File: packages/react-ui/src/theme/carto-theme.js (hunk 2 above)
Comment (vmilan) [reply packages/react-ui/src/theme/carto-theme.js:61]: Fine for me. I've just leave a comment on Figma suggesting this change on their side too, to use de same naming.

File: packages/react-ui/src/theme/sections/palette.js (hunk 3 above)
Comment (vmilan) [reply packages/react-ui/src/theme/sections/palette.js:183]: Yep, typo. This review was just a draft, so I'd planned to do a final check before submitting it for real review, but good catch, thanks :)

File: packages/react-ui/src/theme/sections/typography.js [hunk 5]
```diff
@@ -111,6 +111,10 @@ const customTypography = {
     ...baseTypography.body1,
     fontStyle: 'italic'
   },
+  body1Medium: {
```
Comment (vmilan) [inline packages/react-ui/src/theme/sections/typography.js:114]: I'm going to remove most of them in the branch with the new Typography component.

File: packages/react-ui/src/theme/carto-theme.js (hunk 2 above)
Comment (vmilan) [reply packages/react-ui/src/theme/carto-theme.js]: Done, they already did the change on their side too

File: packages/react-ui/storybook/stories/common/Palette.stories.js (hunk 4 above)
Comment (vmilan) [reply packages/react-ui/storybook/stories/common/Palette.stories.js]: Changed, thanks for the suggestion!

File: UPGRADE.md [hunk 6]
```diff
@@ -33,11 +36,17 @@ Some keys have been removed from [color palette](https://github.com/CartoDB/cart
...
-Some others have been moved because they aren't native MUI keys and are so specific to some components, these are:
+We also have a set of `shade` colors (with transparency):
 
-- charts
-- other
+- black
+- white
 
-These sets of keys are now in: `componentsPalette`.
+Important: `primary.relatedLight` and `secondary.relatedLight` has to be replaced by `primary.background` and `secondary.background`.
```
Comment (VictorVelarde) [inline UPGRADE.md:52]: Should we start doing it in the code, after landing this PR? or not yet?

Comment (vmilan) [reply UPGRADE.md:52]: I already did it here in c4r, in this PR. I left this note for other platforms like cloud-native.

Comment (VictorVelarde) [inline UPGRADE.md:52]: I think that these details are fine, cause are useful for the 'UPGRADE' process itself, but we have also some "mixed" comments (?) about internal changes / refactors (a bit at the beginning of the doc?)... Maybe it's easier to have a separated section in the document talking about 'internals' (some notes useful for carto-react developers) vs 'upgrade' advices for carto-react users (just interested in external / api changes). Just a feeling, not something urgent now, but to think about

File: packages/react-ui/src/theme/carto-theme.d.ts [hunk 7]
```diff
@@ -8,17 +8,68 @@ import { Palette, PaletteColor } from '@mui/material'
...
+  {
+    relatedDark: string;
+    relatedLight: string;
+    background: string;
+    outlinedBorder: string;
+    hint: string;
+    navyBlue: string;
+    locationRed: string;
+    predictionBlue: string;
+    softBlue: string;
```
Comment (VictorVelarde) [inline packages/react-ui/src/theme/carto-theme.d.ts:23]: Are different palettes gonna have variations of these props? It feels a bit strange. I don't want to block it but we should discuss about it

File: UPGRADE.md (hunk 6 above)
Comment (vmilan) [reply UPGRADE.md:52]: Oh, good point. I'll give it a thought to see how to apply it to the document. Thanks for the suggestion :)

File: packages/react-ui/src/theme/carto-theme.d.ts (hunk 7 above)
Comment (vmilan) [reply packages/react-ui/src/theme/carto-theme.d.ts:28]: No, I added it this way to simplify, but you're right. I've just updated this file to be more granular. Let me know if it's ok.

File: packages/react-ui/src/theme/themeUtils.js [hunk 8]
```diff
@@ -10,3 +10,16 @@ export function getPixelToRem(px) {
...
+
+// Lighten and darken color design formula.
+// The Mui lighten / darken function doesn't get the same color as designers use
+export function getMixedColor(color, baseColor, alpha) {
+  const getRgbaBaseColor = (baseColor, alpha = 1) => {
+    const [r, g, b] = baseColor.match(/\w\w/g).map((x) => parseInt(x, 16));
+    return `rgba(${r},${g},${b},${alpha})`;
+  };
+  const rgbaBaseColorValue = getRgbaBaseColor(baseColor, alpha);
+  const colorResult = `linear-gradient(0deg, ${rgbaBaseColorValue}, ${rgbaBaseColorValue}), ${color}`;
```
Comment (VictorVelarde) [inline packages/react-ui/src/theme/themeUtils.js:22]: Ok, I understand you have already validated result with Design. If result is ok for them, I'm ok

File: packages/react-ui/src/theme/carto-theme.d.ts [hunk 9]
```diff
@@ -9,20 +9,44 @@ import { Palette, PaletteColor } from '@mui/material';
...
   PaletteColor,
   {
     relatedDark: string;
     relatedLight: string;
     background: string;
     outlinedBorder: string;
-    hint: string;
+  }
+>;
+type CustomBrandPaletteColor = Modify<
```
Comment (VictorVelarde) [inline packages/react-ui/src/theme/carto-theme.d.ts:22]: 👍🏻 Yes, I think that makes more sense... don't you?

Comment (vmilan) [reply packages/react-ui/src/theme/carto-theme.d.ts:22]: Yep, my bad for trying to go faster. Next time I'll do it this way.

File: packages/react-ui/src/theme/themeUtils.js (hunk 8 above)
Comment (vmilan) [reply packages/react-ui/src/theme/themeUtils.js:22]: I've updated vercel again to let them validate it.

**Pull request merged**